sal-common/src/main/java/org/ow2/proactive/sal/model/Deployment.java

```diff
@@ -103,6 +108,32 @@ public Node getNode() {
         }
     }
 
+    public static Deployment fromId(String nodeName) {
+        Deployment deployment = null;
+        ApplicationContext applicationContext = SpringConfiguration.contextProvider().getApplicationContext();
+        if (applicationContext != null) {
+            DeploymentRepository deploymentRepository = (DeploymentRepository) applicationContext.getBean("deploymentRepository");
+            deployment = deploymentRepository.findOne(nodeName);
+        }
+        if (deployment == null) {
+            deployment = new Deployment();
+            deployment.nodeName = nodeName;
+        }
+        return deployment;
+    }
```

Comment:
I think this method should be in the `RepositoryService `class and not static.

Also you don't need to manually call `applicationContext.getBean();`.
`@Repository` annoted classes can be `@Autowired` in other classes.
If you don't want to autowire the field and keep it like this, you can add a parameter to getBean to specify the returned class type so that you don't need to manually cast. `DeploymentRepository deploymentRepository = applicationContext.getBean("deploymentRepository", DeploymentRepository.class);`